lib/pdfh/models/document_type.rb

```diff
@@ -62,5 +68,14 @@ def extract_subtype(sub_types)
         DocumentSubType.new(name: name, month_offset: offset)
       end
     end
+
+    # @raise [ArgumentError] when called
+    # @return [void]
+    def raise_validators_error
+      template = "has invalid %<1>s. Unknown tokens: %<2>s"
+      path_errors = @path_validator.valid? ? "" : format(template, :store_path, @path_validator.unknown.join(", "))
+      name_errors = @name_validator.valid? ? "" : format(template, :name_template, @name_validator.unknown.join(", "))
+      raise ArgumentError, "Document type #{name.inspect} #{path_errors} #{name_errors}"
+    end
```

Comment:
You know that I like ternaries,  but not sure about the empty string when there're not errors. It's to avoid an NameError when you do string interpolation below, isn't it? But if you define the variable as nil, it won't cause NameError. So I think that it would be more rubyist if you try:

```ruby
path_errors = format(template, :store_path, @path_validator.unknown.join(", ")) unless @path_validator.valid?
name_errors = format(template, :name_template, @name_validator.unknown.join(", ")) unless @name_validator.valid?
```

Also, both definitions follows same pattern, so I guess that you could define a method for a dryer code. Maybe

```ruby
def template_error_message(error_type, validator_unknown)
  template = "has invalid %<1>s. Unknown tokens: %<2>s"
  
  format(template, error_type, validator_unknown
end
 ```
 
 Unless you think is too much just for two cases.